qemu/tests/qemu_guest_agent.py

```diff
@@ -2588,6 +2588,162 @@ def bl_check(qga_cmd):
+        timezone_ga = self.gagent.get_timezone()
+        timezone_name_ga = timezone_ga["zone"]
+        timezone_offset_ga = timezone_ga["offset"]
+
+        logging.info("Check timezone name.")
+        cmd_get_timezone_name = params["cmd_get_timezone_name"]
+        timezone_name_guest = session.cmd_output(cmd_get_timezone_name).\
+            strip()
+        if params["os_type"] == "windows":
+            # there are standard name and daylight name for windows os
+            # both are accepted.
+            cmd_dlight_name = params["cmd_get_timezone_dlight_name"]
+            timezone_dlight_name_guest = session.cmd_output(cmd_dlight_name).\
+                strip()
+            timezone_name_list = [timezone_name_guest,
+                                  timezone_dlight_name_guest]
+            if timezone_name_ga not in timezone_name_list:
+                msg = "The result is different between qga and guest\n"
+                msg += "from qga: %s\n" % timezone_name_ga
+                msg += "from guest: %s\n" % timezone_name_list
+                test.fail(msg)
+        else:
+            _result_check(timezone_name_ga, timezone_name_guest)
+
+        logging.info("Check timezone offset.")
+        cmd_get_timezone_offset = params["cmd_get_timezone_offset"]
+        timezone_offset_guest = session.cmd_output(cmd_get_timezone_offset).\
+            strip()
+        # +08:00
+        # (UTC+08:00) Beijing, Chongqing, Hong Kong, Urumqi
+        pattern = r"(\S)(\d\d):\d\d"
+        timezone_list = re.findall(pattern, timezone_offset_guest, re.I)
+        # if it's daylight save time, offset should be 1h early
+        if "daylight" in timezone_name_ga.lower():
+            offset_seconds = (int(timezone_list[0][1]) - 1) * 3600
+        else:
+            offset_seconds = int(timezone_list[0][1]) * 3600
+        if timezone_list[0][0] == "-":
+            timezone_offset_guest_seconds = int(timezone_list[0][0]
+                                                + str(offset_seconds))
+        else:
+            timezone_offset_guest_seconds = int(offset_seconds)
+        _result_check(timezone_offset_ga, timezone_offset_guest_seconds)
+
+        error_context.context("Check the current active users number.",
+                              logging.info)
+        user_qga_list = self.gagent.get_users()
+        user_num_qga = len(user_qga_list)
+        cmd_get_users = params["cmd_get_users"]
+        user_guest = session.cmd_output(cmd_get_users).strip()
+        user_guest_list = user_guest.split("\n")
+
+        logging.info("Get all users name in guest.")
+        if params["os_type"] == "linux":
+            cmd_get_user_name = params["cmd_get_users_name"]
+            user_name_guest = session.cmd_output(cmd_get_user_name).strip()
+            user_name_list_guest = user_name_guest.split("\n")
+        else:
+            user_name_list_guest = []
+            for user in user_guest_list:
+                user = user.strip().strip('>')
```

Comment:
```
user = user.strip(' >')
```

Reply:
I think user = user.strip().strip('>') is okay, it can also clear the space at the end.

Reply:
you could do both in one call, the string in my demo is `space` + `>`.

Reply:
you are right, will modify.